quinn/src/streams.rs

```diff
@@ -428,7 +439,27 @@ where
         cx: &mut Context,
         bufs: &mut [Bytes],
     ) -> Poll<Result<Option<usize>, ReadError>> {
-        self.poll_read_generic(cx, |conn, stream| conn.inner.read_chunks(stream, bufs))
+        self.poll_read_generic(cx, true, |chunks| {
+            let mut read = 0;
+            loop {
+                if read >= bufs.len() {
+                    return match read {
+                        0 => Ok(None),
```

Comment:
Would this cause us to return `Ok(None)` for a non-finished stream when `bufs.len() == 0`?

Reply:
Yeah, I wasn't sure how to handle the case where `bufs.len() == 0`. Maybe that needs a new `ReadError` variant?

Reply:
What's wrong with `Ok(Some(0))`?

Reply:
Yeah, I guess that is correct.